app/models/metrics.rb

```diff
@@ -33,22 +33,22 @@ def initialize(root, group_by: nil, time_period:)
   alias :read_attribute_for_serialization :send
   attr_reader :group_by, :root, :time_period
 
-  def metrics
+  def metrics(entity: root)
     [
-      AggregatedCallsReceivedMetric.new(root, time_period),
-      AggregatedTransactionsReceivedMetric.new(root, time_period),
-      AggregatedTransactionsWithOutcomeMetric.new(root, time_period)
-    ]
+      AggregatedCallsReceivedMetric,
+      AggregatedTransactionsReceivedMetric,
+      AggregatedTransactionsWithOutcomeMetric
+    ].map { |metric|
```

Comment:
Not sure I agree that iterating the classes and then instantiating is much cleaner, than just instantiating them inline. Regardless, I'd combine the conditional and `compact` to a `select(&:is_applicable?)`.